src/Lifecycle/StateManager.php

```diff
@@ -188,18 +200,47 @@ protected function loadMany(iterable $ids, string $type): StateCollection
     protected function reconstitute(State $state): static
     {
         // When we're replaying, the Broker is in charge of applying the correct events
-        // to the State, so we only need to do it *outside* of replays.
-        if (! $this->is_replaying) {
-            $this->events
-                ->read(state: $state, after_id: $state->last_event_id)
-                ->each(fn (Event $event) => $this->dispatcher->apply($event));
-
-            // It's possible for an event to mutate state out of order when reconstituting,
-            // so as a precaution, we'll clear all other states from the store and reload
-            // them from snapshots as needed in the rest of the request.
-            // FIXME: We still need to figure this out
-            // $this->states->reset();
-            // $this->remember($state);
+        // to the State, so we need to skip during replays. Similarly, if we're already
+        // reconstituting in a recursive call, the root call is responsible for applying
+        // events, so we should also skip in that case.
+
+        if (! $this->is_replaying && ! $this->is_reconstituting) {
+            try {
+                $this->is_reconstituting = true;
+
+                $summary = $this->events->summarize($state);
+
+                // FIXME: We probably need to re-write all the snapshots after we're done
+                // FIXME: Swap out existing state manager, push all related states into new state manager
+                // FIXME: run all the event on them, swap them out
```

Comment:
This is the correct approach, I think. Ultimately, we just need an isolated state manager to handle everything and then pull those values over into the global state manager after it's done.